core/os/os.cpp

```diff
@@ -290,6 +290,29 @@ String OS::get_system_dir(SystemDir p_dir, bool p_shared_storage) const {
 	return ".";
 }
 
+String OS::system_dir_to_string(SystemDir p_dir) {
+	switch (p_dir) {
+		case SYSTEM_DIR_DESKTOP:
+			return "SYSTEM_DIR_DESKTOP";
```

Comment:
Can't we let them all fall through if it matches any case and just `return p_dir` in last case?
That way we can only have default case different.

Reply:
I think you're misunderstanding what that function does. It is only there to provide a String representation of the enum for use in the error message. There doesn't seem to be a good function to stringify any enum without importing an unnecessary library into the engine, so I added system_dir_to_string.

Reply:
My bad. I didn't notice p_dir was an enum not a string.